src/ray/gcs/gcs_server/test/gcs_server_rpc_test.cc

```diff
@@ -1,88 +1,189 @@
-#include <boost/asio/basic_socket.hpp>
-#include "gmock/gmock.h"
 #include "gtest/gtest.h"
+#include "ray/gcs/gcs_server/actor_info_handler_impl.h"
 #include "ray/gcs/gcs_server/gcs_server.h"
+#include "ray/gcs/gcs_server/job_info_handler_impl.h"
 #include "ray/rpc/gcs_server/gcs_rpc_client.h"
 
 namespace ray {
-class GcsServerRpcTest : public ::testing::Test {
- protected:
-  template <typename T>
-  class MockedGcsServer : public gcs::GcsServer {
-   public:
-    explicit MockedGcsServer(const gcs::GcsServerConfig &config)
-        : gcs::GcsServer(config) {}
+class GcsServerTest : public ::testing::Test {
+ public:
+  GcsServerTest() : options_("127.0.0.1", 6379, "", true) {}
+
+  void SetUp() override {
+    thread_.reset(new std::thread([this] {
+      std::unique_ptr<boost::asio::io_service::work> work(
+          new boost::asio::io_service::work(io_service_));
+      io_service_.run();
+    }));
+
+    // create redis gcs client
+    redis_gcs_client_ = std::make_shared<gcs::RedisGcsClient>(options_);
+    RAY_CHECK_OK(redis_gcs_client_->Connect(io_service_));
+
+    // create gcs rpc server
+    server_.reset(new rpc::GrpcServer("gcs_server", 0));
+    job_info_handler_.reset(new rpc::DefaultJobInfoHandler(*redis_gcs_client_));
+    job_info_service_.reset(new rpc::JobInfoGrpcService(io_service_, *job_info_handler_));
+    actor_info_handler_.reset(new rpc::DefaultActorInfoHandler(*redis_gcs_client_));
+    actor_info_service_.reset(
+        new rpc::ActorInfoGrpcService(io_service_, *actor_info_handler_));
+    server_->RegisterService(*job_info_service_);
+    server_->RegisterService(*actor_info_service_);
+    server_->Run();
 
-   protected:
-    void InitBackendClient() override {}
+    // create gcs rpc client
+    client_call_manager_.reset(new rpc::ClientCallManager(io_service_));
+    client_.reset(
+        new rpc::GcsRpcClient("0.0.0.0", server_->GetPort(), *client_call_manager_));
```

Comment:
Why not use `gcs_server` directly?

Reply:
Use gcs_server directly is a good idea, i have fix it.